### ISSUE TITLE
Unable to retrieve Arn attribute for AWS::ECR::Repository, with error

### ISSUE THREAD

**Issue by scble** [aws/copilot-cli#5683]
-----
Environment
- copilot version: v1.33.1
- Cluster: Fargate
----

I tried to deploy a new svc.
I already have 3 environments and 3 services running in each environment.

However, the StackSet-xxxxx-web-infrastructure stack（Description: Cross-regional resources to support the CodePipeline for a workspace） has become UPDATE_ROLLBACK_COMPLETE.

>copilot svc delete --name example-app 

I tried to delete, but get the error 

>"Unable to retrieve Arn attribute for AWS::ECR::Repository, with error".

I then tried to run delete again, but get the following message.
"✘ couldn't find service example-app in the application xxxxx".

copilot svc init --name example-app 
I tried executing
```
Resource handler returned message: "Resource of type 'AWS::ECR::Reposi
      tory' with identifier 'xxxxxxx/example-app' already exists.
      " (RequestToken: xxxxxxxxxxxxxxxx, HandlerErrorCod
      e: AlreadyExists)
```

I ran init again and it recommended deploying as follows.
```
Run `copilot svc deploy --name example-app --env test` to deploy your service to a test environment.
```

Therefore, I ran deploy but it stopped with the following error.
```
 - An ECS task definition to group your containers and run them on ECS              [delete complete]    [3.8s]
    Resource handler returned message: "Invalid request provided: Create T
    askDefinition: Container.image repository should not be null or empty.
     (Service: AmazonECS; Status Code: 400; Error Code: ClientException; R
    equest ID: 6be2f88f-c21a-4546-8984-806e9a7bfb38; Proxy: null)" (Reques
    tToken: 0932c9f5-648b-8296-b32f-3adb6d795cbe, HandlerErrorCode: Invali
    dRequest)
```

✅  I checked the stack parameters and found that ContainerImage is specified as a URI in the stack of other successful services, but the failing task has a digest (@sha256:xxxxxxxxxxxxxxxxxxxxxxxxx).

I also ran copilot svc delete and was able to confirm that the service stack was deleted, but I got the following error and the stack for StackSet-xxxx-web-infrastructure was UPDATE_ROLLBACK_IN_ with the following error PROGRESS.
>Unable to retrieve Arn attribute for AWS::ECR::Repository, with error message null

Is there any way to recover?

ECR Repository permissions and tags are set.

**Comment by scble** [comment]
I found that the ECR does not have the following permissions and tags.
I thought that when I created it with copilot svc init, the repository was automatically created as well, but was I wrong? I don't remember putting the policy here myself before...

Permission
====
ecr:BatchCheckLayerAvailability
ecr:BatchGetImage
ecr:CompleteLayerUpload
ecr:GetDownloadUrlForLayer
ecr:InitiateLayerUpload
ecr:PutImage
ecr:UploadLayerPart
====

repository tags
- copilot-application
- copilot-service

Still, it was not resolved. I am getting a similar error.
>Invalid request provided: Create TaskDefinition: Container.image repository should not be null or empty.

**Comment by scble** [comment]
copilot svc init --name example-app
I have confirmed that the ECR has been created after executing the following command, and that it has permissions and tags. However, I then received the following error and confirmed that the repository had been deleted.

>Unable to retrieve Arn attribute for AWS::ECR::Repository, with error

Therefore, when we look at the resources and output of the stack, we do not recognize the example-app repository, so we cannot proceed with the deployment either.

**Comment by scble** [comment]
https://github.com/aws/copilot-cli/issues/4296
I don't want to delete the stack because it contains the production environment settings.

**Comment by scble** [comment]
Sorry, I found the cause.
I checked the corresponding stack template from CloudFormation's StackSets and found that there are multiple RepositoryName's.
This is as expected.

However, a non-existent ECR repository configuration remained. It was probably deleted manually.

In other words, the problem was not with the newly deployed workloads I wanted to deploy, but with the ECR repository not existing, even though the previously existing ECR configuration was still there.
